src/Symfony/Component/Security/Core/Authorization/Voter/TraceableVoter.php

```diff
@@ -37,6 +37,13 @@ public function vote(TokenInterface $token, $subject, array $attributes): int
     {
         $result = $this->voter->vote($token, $subject, $attributes);
 
+        if (false === \in_array($result, [self::ACCESS_DENIED, self::ACCESS_GRANTED, self::ACCESS_ABSTAIN], true)) {
+            @trigger_error(
+                sprintf('Return value of "%s::vote()" should always be of the type int since Symfony 5.0, %s returned.', \get_class($this->voter), \gettype($result)),
```

Comment:
We should use `get_debug_type()` both times here.
I agree also that this should use `is_int()`, as `in_array()` is stricter than what the message suggests.

Reply:
get_debug_type in introduced in PHP 8.0. We speak about 5.x branch that is supposed to support php 7.2.5
in_array with strict type set to "true" is as strict as "is_int.

however taking into account what @wouterj wrote "...but it's about being able to simulate the : int return type" - it is better to use is_int.

Reply:
get_debug_type is still available, thanks to the php8 polyfill that the component already requires